--- ./src/promisc.c.orig	Sun Jun  3 02:20:18 2001
+++ ./src/promisc.c	Sun Jun  3 02:21:55 2001
@@ -80,6 +80,7 @@
 	     */
 
 	    if ((strncmp(buf, "eth", 3) == 0) ||
+		(strncmp(buf, "wvlan", 5) == 0) ||
 		(strncmp(buf, "fddi", 4) == 0)) {
 		strcpy(ifr.ifr_name, buf);
 
@@ -190,6 +191,7 @@
 
     while (ptmp != NULL) {
 	if (((strncmp(ptmp->params.ifname, "eth", 3) == 0) ||
+	     (strncmp(ptmp->params.ifname, "wvlan", 5) == 0) ||
 	     (strncmp(ptmp->params.ifname, "fddi", 4) == 0)) &&
 	    (ptmp->params.state_valid)) {
 
--- ./src/packet.c.orig	Sun Jun  3 02:22:41 2001
+++ ./src/packet.c	Sun Jun  3 02:24:26 2001
@@ -66,11 +66,13 @@
     switch (family) {
     case ARPHRD_ETHER:
 	if (strncmp(ifname, "eth", 3) == 0)
-	    result = LINK_ETHERNET;
+		result = LINK_ETHERNET;
+	else if (strncmp(ifname, "wvlan", 5) == 0)
+		result = LINK_ETHERNET;
 	else if (strncmp(ifname, "plip", 4) == 0)
-	    result = LINK_PLIP;
+		result = LINK_PLIP;
 	else if (strncmp(ifname, "fddi", 4) == 0)	/* For some Ethernet */
-	    result = LINK_ETHERNET;	/* emulated FDDI ifaces */
+		result = LINK_ETHERNET;			/* emulated FDDI ifaces */
 	else if (strncmp(ifname, "dvb", 3) == 0)
 	    result = LINK_ETHERNET;
 	else if ((strncmp(ifname, "isdn", 4) == 0) && (isdn_fd != -1)) {
--- ./src/rvnamed.c.orig	Sun Jun  3 02:24:48 2001
+++ ./src/rvnamed.c	Sun Jun  3 02:26:34 2001
@@ -210,6 +210,9 @@
     signal(SIGCHLD, childreap);
 
     logfile = fopen(RVNDLOGFILE, "a");
+    if ( logfile == NULL ) { /* unable to open file -> /dev/null */
+	    logfile = fopen("/dev/null", "a");
+    }
 
     writervnlog(logfile, "******** rvnamed started ********");
     writervnlog(logfile, "Clearing socket names");
--- iptraf-2.5.0/src/ifaces.c.orig	Thu Jun  7 05:27:42 2001
+++ iptraf-2.5.0/src/ifaces.c	Wed Aug 15 02:03:07 2001
@@ -31,13 +31,13 @@
 #include "error.h"
 #include "links.h"
 
-#define NUM_SUPPORTED_IFACES 11
+#define NUM_SUPPORTED_IFACES 12
 
 extern int daemonized;
 
 char ifaces[NUM_SUPPORTED_IFACES][6] =
     { "lo", "eth", "sl", "ppp", "ippp", "plip", "fddi", "isdn", "dvb",
-      "pvc", "hdlc" };
+      "pvc", "hdlc", "wvlan" };
 
 char *ltrim(char *buf)
 {
