diff -Nur iptraf-2.6.0.org/Documentation/iptraf.8 iptraf-2.6.0/Documentation/iptraf.8
--- iptraf-2.6.0.org/Documentation/iptraf.8	Mon Mar 18 09:05:19 2002
+++ iptraf-2.6.0/Documentation/iptraf.8	Wed May  1 13:30:34 2002
@@ -86,7 +86,7 @@
 
 .SH FILES
  /var/log/iptraf/*.log - log file
- /var/local/iptraf/* - important IPTraf data files
+ /var/lib/iptraf/* - important IPTraf data files
 
 .SH SEE ALSO
  Documentation/* - complete documentation written by the author
diff -Nur iptraf-2.6.0.org/Documentation/rvnamed.8 iptraf-2.6.0/Documentation/rvnamed.8
--- iptraf-2.6.0.org/Documentation/rvnamed.8	Mon Mar 13 13:03:59 2000
+++ iptraf-2.6.0/Documentation/rvnamed.8	Wed May  1 13:30:34 2002
@@ -11,7 +11,7 @@
 This program is only used by iptraf and, therefore, is useless alone.
 
 .SH FILES
- /var/local/iptraf/rvnamed.log - log file
+ /var/log/iptraf/rvnamed.log - log file
 
 .SH SEE ALSO
 README.rvnamed - documentation from the author
diff -Nur iptraf-2.6.0.org/src/Makefile iptraf-2.6.0/src/Makefile
--- iptraf-2.6.0.org/src/Makefile	Wed Apr 24 05:30:44 2002
+++ iptraf-2.6.0/src/Makefile	Wed May  1 13:31:02 2002
@@ -32,7 +32,7 @@
 # options to be passed to the compiler.  I don't believe they need to be
 # modified (except for -m486 on non-Intel x86 platforms).
 
-CFLAGS		= -Wall -O2 #-m486
+CFLAGS		= $(RPM_OPT_FLAGS)
 DIRS		= -DWORKDIR=\"$(WORKDIR)\" \
 		  -DLOGDIR=\"$(LOGDIR)\" -DEXECDIR=\"$(TARGET)\"
 LDOPTS		= #-static
@@ -69,12 +69,12 @@
 # installation target directory.  The iptraf and rvnamed programs get
 # stored here.  iptraf also exec's rvnamed from this directory.
 
-TARGET		= /usr/local/bin
+TARGET		= /usr/bin
 
 # The IPTraf working directory; if you change this.  Starting with this
 # version, this definition overrides dirs.h.
 
-WORKDIR		= /var/local/iptraf
+WORKDIR		= /var/lib/iptraf
 
 # The IPTraf log file directory.  IPTraf log files are placed here.
 # This definition overrides dirs.h
@@ -90,7 +90,7 @@
 # DO NOT MAKE THIS POINT TO AN EXISTING/SYSTEM DIRECTORY!!!!  THE
 # LOCK OVERRIDE (iptraf -f) WILL ERASE FILES HERE!
 #*****************************************************************
-LOCKDIR		= /var/run/iptraf
+LOCKDIR		= /var/run
 
 #####################################################################
 ############### IPTRAF COMPILATION AND LINK RULES
diff -Nur iptraf-2.6.0.org/src/dirs.h iptraf-2.6.0/src/dirs.h
--- iptraf-2.6.0.org/src/dirs.h	Thu May  3 12:44:11 2001
+++ iptraf-2.6.0/src/dirs.h	Wed May  1 13:30:34 2002
@@ -26,7 +26,7 @@
  */
  
 #ifndef WORKDIR
-#define WORKDIR		"/var/local/iptraf"
+#define WORKDIR		"/var/lib/iptraf"
 #endif
 
 #ifndef LOGDIR
@@ -34,7 +34,7 @@
 #endif
 
 #ifndef EXECDIR
-#define EXECDIR		"/usr/local/bin"
+#define EXECDIR		"/usr/bin"
 #endif
 
 /*
@@ -46,7 +46,7 @@
  */
  
 #ifndef LOCKDIR
-#define LOCKDIR		"/var/run/iptraf"
+#define LOCKDIR		"/var/run"
 #endif
 
 /***
