--- iptraf-3.0.0/src/ifaces.c.orig	2007-06-03 19:03:27.312372537 +0000
+++ iptraf-3.0.0/src/ifaces.c	2007-06-03 19:07:20.170408147 +0000
@@ -32,14 +32,14 @@
 #include "error.h"
 
 extern int accept_unsupported_interfaces;
-#define NUM_SUPPORTED_IFACES 26
+#define NUM_SUPPORTED_IFACES 28
 
 extern int daemonized;
 
 char ifaces[][6] =
     { "lo", "eth", "sl", "ppp", "ippp", "plip", "fddi", "isdn", "dvb",
     "pvc", "hdlc", "ipsec", "sbni", "tr", "wvlan", "wlan", "sm2", "sm3",
-    "pent", "lec", "brg", "tun", "tap", "cipcb", "tunl", "vlan", "ath"
+    "pent", "lec", "brg", "tun", "tap", "cipcb", "tunl", "vlan", "bond", "ath"
 };
 
 char *ltrim(char *buf)
--- iptraf-3.0.0/src/packet.c.orig	2007-06-03 19:03:27.312372537 +0000
+++ iptraf-3.0.0/src/packet.c	2007-06-03 19:03:27.316372709 +0000
@@ -91,7 +91,8 @@
         else if (strncmp(ifname, "ipsec", 5) == 0)
             result = LINK_ETHERNET;
         else if ((strncmp(ifname, "wvlan", 5) == 0)
-                 || (strncmp(ifname, "wlan", 4) == 0))
+                 || (strncmp(ifname, "wlan", 4) == 0)
+		 || (strncmp(ifname, "ath", 3) == 0))
             result = LINK_ETHERNET;
         else if ((strncmp(ifname, "sm2", 3) == 0)
                  || (strncmp(ifname, "sm3", 3) == 0))
