--- iptraf-2.7.0/src/tcptable.h.orig	2005-05-09 11:49:03.000000000 +0200
+++ iptraf-2.7.0/src/tcptable.h	2005-05-09 11:50:28.000000000 +0200
@@ -12,19 +12,17 @@
 #include <fcntl.h>
 #include <curses.h>
 #include <panel.h>
-#include <netdb.h>
 #include <time.h>
 #include <sys/types.h>
-#include <sys/socket.h>
-#include <netinet/in.h>
-#include <arpa/inet.h>
-#include <net/if_arp.h>
 #include <asm/types.h>
+#include <linux/socket.h>
+#include <sys/socket.h>
+#include <linux/net.h>
+#include <linux/if.h>
 #include <linux/if_packet.h>
 #include <linux/if_ether.h>
 #include <linux/if_fddi.h>
 #include <linux/if_tr.h>
-#include <net/if.h>
 #include <netinet/ip.h>
 #include <netinet/udp.h>
 #include "servname.h"
--- iptraf-3.0.0.orig/src/othptab.c
+++ iptraf-3.0.0/src/othptab.c
@@ -16,14 +16,28 @@
 
 ***/
 
+/* ARP protocol opcodes. */
+#define ARPOP_REQUEST   1               /* ARP request.  */
+#define ARPOP_REPLY     2               /* ARP reply.  */
+#define ARPOP_RREQUEST  3               /* RARP request.  */
+#define ARPOP_RREPLY    4               /* RARP reply.  */
+#define ARPOP_InREQUEST 8               /* InARP request.  */
+#define ARPOP_InREPLY   9               /* InARP reply.  */
+#define ARPOP_NAK       10              /* (ATM)ARP NAK.  */
+
+
 #include <asm/types.h>
+#include "options.h"
+#include "tcptable.h"
+/*#include <linux/socket.h>*/
+#include <linux/if.h>
 #include <linux/if_ether.h>
 #include <linux/if_tr.h>
 #include <linux/if_fddi.h>
+#include <linux/if_arp.h>
+#include <netdb.h>
 #include <winops.h>
 #include "arphdr.h"
-#include "options.h"
-#include "tcptable.h"
 #include "othptab.h"
 #include "deskman.h"
 #include "attrs.h"
